src/models/InviteCode.ts

```diff
@@ -1,7 +1,27 @@
 import { HexString } from '../helpers/opaqueTypes';
 
+export const enum InviteCodeFields {
+    version = 'version',
+    randomSeed = 'randomSeed',
+    contactPublicKey = 'contactPublicKey',
+    profileName = 'profileName',
+    expiry = 'expiry',
+}
+
+export const INVITE_CODE_VERSION = 1;
+
 export interface InviteCode {
-    randomSeed: HexString;
-    contactPublicKey: HexString;
-    profileName?: string;
+    [InviteCodeFields.version]: number;
+    [InviteCodeFields.randomSeed]: HexString;
+    [InviteCodeFields.contactPublicKey]: HexString;
+    [InviteCodeFields.profileName]: string;
+    [InviteCodeFields.expiry]: number;
 }
+
+export const isVersion1RawInviteCode = (jsonObj: any): boolean => {
```

Comment:
@agazso there will be a better solution for this in TypeScript 3.7, called `assert signatures`. If you check the call site you can see that it would make sense to throw the error from such function. 
https://httptoolkit.tech/blog/5-big-features-of-typescript-3.7/